test/FunctionalTests/Microsoft.OData.Core.Tests/AsyncYieldStream.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Tests
+{
+    /// <summary>
+    /// Stream wrapper that yields in every async method
+    /// to ensure the read/write operations complete asynchronously
+    /// even when the operate completely in-memory. This helps simulate
+    /// async I/O in unit tests without making requests out of the process.
+    /// </summary>
+    internal class AsyncYieldStream : Stream
+    {
+        private Stream stream;
+        public AsyncYieldStream(Stream stream)
+        {
+            this.stream = stream;
+        }
+
+        public override bool CanRead => this.stream.CanRead;
+
+        public override bool CanSeek => this.stream.CanSeek;
+
+        public override bool CanWrite => this.stream.CanWrite;
+
+        public override long Length => this.stream.Length;
+
+        public override int ReadTimeout { get => this.stream.ReadTimeout; set => this.stream.ReadTimeout = value; }
+
+        public override int WriteTimeout { get => this.stream.WriteTimeout; set => this.stream.WriteTimeout = value; }
+
+        public override bool CanTimeout => this.stream.CanTimeout;
+
+        public override void Close()
+        {
+            this.stream.Close();
+        }
+
+        public override long Position { get => this.stream.Position; set => this.stream.Position = value; }
+
+        public override void Flush()
+        {
+            this.stream.FlushAsync().Wait();

```

Comment:
Would it be more appropriate to call `this.stream.Flush()` instead of blocking on the async method?

Reply:
resolved with `this.stream.Flush`